### ISSUE TITLE
zig fmt: comptime not allowed inside struct

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#2260]
```zig
pub fn GeneralPurposeDebugAllocator(comptime config: Config) type {
    return struct {
        allocator: Allocator,

        comptime {
            if (config.backing_allocator and config.memory_protection) {
                @compileError("Memory protection is unavailable when using a backing allocator");
            }
        }
    };
}
```

Expected: zig fmt accepts the input
Actual:`error: Expected variable declaration or function, found Keyword_comptime`

**Comment by mikdusan** [comment]
also top-level fields not allowed:

#### ftest.zig
```zig
// ok
const Foo = struct {
    a: usize,
};

// error
b: usize,
```

#### `zig fmt ftest.zig`
```
ftest.zig:7:1: error: Expected variable declaration or function, found Identifier
b: usize,
```

**Comment by emekoi** [comment]
@andrewrk closed by #2308.

**Comment by andrewrk** [comment]
@mikdusan see #2022  for top level fields
